apps/api/v2/src/modules/workflows/workflows.repository.ts

```diff
@@ -0,0 +1,305 @@
+export class WorkflowsRepository {
+  constructor(
+    private readonly dbRead: PrismaReadService,
+    private readonly dbWrite: PrismaWriteService,
+    private readonly teamsVerifiedResourcesRepository: TeamsVerifiedResourcesRepository
+  ) {}
+
+  async deleteTeamWorkflowById(teamId: number, workflowId: number) {
+    return await this.dbWrite.prisma.workflow.delete({ where: { id: workflowId, teamId } });
+  }
+
+  async getTeamWorkflowById(teamId: number, id: number): Promise<WorkflowType> {
+    const workflow = await this.dbRead.prisma.workflow.findUnique({
+      where: {
+        id: id,
+        teamId: teamId,
+      },
+      include: {
+        steps: true,
+        activeOn: { select: { eventTypeId: true } },
+      },
+    });
+
+    if (!workflow) {
+      throw new NotFoundException(`Workflow with ID ${id} not found for team ${teamId}`);
+    }
+
+    return workflow;
+  }
+
+  async getTeamWorkflows(teamId: number, skip: number, take: number): Promise<WorkflowType[]> {
+    const workflows = await this.dbRead.prisma.workflow.findMany({
+      where: {
+        teamId: teamId,
+      },
+      include: {
+        steps: true,
+        activeOn: { select: { eventTypeId: true } },
+      },
+      skip,
+      take,
+    });
+
+    if (!workflows?.length) {
+      throw new NotFoundException(`team ${teamId} does not have any Workflows.`);
+    }
+
+    return workflows;
+  }
+
+  async createTeamWorkflowHusk(teamId: number) {
+    return this.dbWrite.prisma.workflow.create({
+      data: {
+        name: "",
+        trigger: WorkflowTriggerEvents.BEFORE_EVENT,
+        time: 24,
+        timeUnit: TimeUnit.HOUR,
+        teamId,
+      },
+      include: { activeOn: true, steps: true },
+    });
+  }
+
+  async createTeamWorkflow(user: UserWithProfile, teamId: number, data: Partial<CreateWorkflowDto>) {
+    const workflowHusk = await this.createTeamWorkflowHusk(teamId);
+
+    await updateWorkflow({
+      ctx: {
+        user: { ...user, locale: user?.locale ?? "en" },
+        prisma: this.dbWrite.prisma as unknown as PrismaClient,
+      },
+      input: await this.mapCreateDtoToZodUpdateSchema(data, workflowHusk.id, teamId, workflowHusk),
+    });
+
+    const workflow = await this.getTeamWorkflowById(teamId, workflowHusk.id);
+    return workflow;
+  }
+
+  async updateTeamWorkflow(
+    user: UserWithProfile,
+    teamId: number,
+    workflowId: number,
+    data: Partial<CreateWorkflowDto>
+  ) {
+    const currentWorkflow = await this.getTeamWorkflowById(teamId, workflowId);
+    await updateWorkflow({
+      ctx: {
+        user: { ...user, locale: user?.locale ?? "en" },
+        prisma: this.dbWrite.prisma as unknown as PrismaClient,
+      },
+      input: await this.mapCreateDtoToZodUpdateSchema(data, workflowId, teamId, currentWorkflow),
+    });
+
+    const workflow = await this.getTeamWorkflowById(teamId, workflowId);
+    return workflow;
+  }
+
+  public toDto(workflow: WorkflowType): WorkflowOutput {
```

Comment:
The `toDto` should be workflows output service because repositories should not know anything about how api is responding.